fleetctl/stop_test.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+
+	"github.com/coreos/fleet/job"
+)
+
+func doStopUnits(r commandTestResults, errchan chan error) {
+	sharedFlags.NoBlock = true
```

Comment:
Nit, this is a side effect. How about doing a sf := sharedFlags.NoBlock; sharedFlags.NoBlock = true; defer func() { sharedFlags.NoBlock = sf } within the context of the test loop?